src/api/exchange/DepositApiImpl.ts

```diff
@@ -0,0 +1,186 @@
+import BN from 'bn.js'
+import assert from 'assert'
+
+import { formatAmount, log, toBN } from 'utils'
+import { ZERO } from 'const'
+import { getEpochAddressForNetwork } from './epochList'
+
+import { DepositApi, Receipt, TxOptionalParams, EpochTokenLocker } from 'types'
+import EpochTokenLockerAbi from './EpochTokenLockerAbi'
+import Web3 from 'web3'
+import { getProviderState, Provider, ProviderState } from '@gnosis.pm/dapp-ui'
+
+const getNetworkIdFromWeb3 = (web3: Web3): null | number => {
+  if (!web3.currentProvider) return null
+
+  // web3.currentProvider may be our provider wrapped in a Proxy
+  // depending on web3 version
+  // or internally created provider from url
+  const providerState: ProviderState | null = getProviderState((web3.currentProvider as unknown) as Provider | null)
+
+  return providerState && providerState.chainId
+}
+
+export class DepositApiImpl implements DepositApi {
+  private _ReferenceEpochTokenLocker: EpochTokenLocker
+  private _web3: Web3
+
+  private static _address2ContractCache: { [K: string]: EpochTokenLocker } = {}
+
+  public constructor(web3: Web3) {
+    this._ReferenceEpochTokenLocker = new web3.eth.Contract(EpochTokenLockerAbi)
+
+    this._web3 = web3
+
+    // TODO remove later
+    ;(window as any).epoch = this._ReferenceEpochTokenLocker
+  }
+
+  public getContractAddress(networkId: number): string | null {
+    return getEpochAddressForNetwork(networkId)
+  }
+
+  public async getBatchTime(): Promise<number> {
+    const contract = await this._getContract()
+    const BATCH_TIME = await contract.methods.BATCH_TIME().call()
+    return +BATCH_TIME
```

Comment:
nit: should this variable be all CAPS?

Reply:
cause the method is in all caps and it's a contract const